docs/assignment2.md

```diff
@@ -1 +1,233 @@
+Each team is an aggregation of exactly 8 **Player** objects.  
+
+#### **League** 
+
+A league simulates multiple rounds of matches between user-selected teams and determines the winners.
+
+Attributes
+
+- *initialTeams* (type:Queue<Team&gt;): List of the teams playing in the first round, as selected by the user.  Already described in **Game**.createLeague's parameter as: list of teams ordered by their playing order, e.g.  a round with matches (Team1 vs. Team2) and (Team3 vs. Team4) would be represented as {Team1, Team2, Team3, Team4}
+-  *winners* (type: HashMap<Integer, ArrayList<Team&gt;>):  A key represents the round number, e.g. first round is 1, while the values represent the winners at each given round.
+
+Operations
+
+- *void playRound(Queue<Team&gt; teamsToPlay, int round)*: recursive method that generates *winners*. First call is *playRound(initialTeams, 1)*. The method calls itself for following rounds until one team remains (the league winner) 
+- HashMap<Integer, ArrayList<Team>> getWinners(): calls *playRound* method and returns *winners*
+
+Associations
+
+A single **League** object can exist at one time within a **Game**. A league can start with at most 16 teams as per requirement **FR6**. Within the *playRound* method, **Match** objects are created from the playing teams to generate a winner.
+
+#### **Match**
+
+A match simulates a single soccer game between two teams.
+
+Attributes
+
+* *team1* (Type: Team): first team in the match
+* *team2* (Type: Team): second team in the match
+
+Operations
+
+- *Team getWinner()*: returns the winner of a match as described in **FR8** and **FR9**
+
+Assiciations
+
+A match is formed of exactly 2 **Team** objects.
+
+### *Views* package
+
+Each class in this package is an extention of the **GamePane** abstract class, described below. Each uses the *javafx* library and represents a different window in the GUI. All interactions with the *models* package are made via the **Game** class methods.
+
+#### **GamePane**
+
+Abstract methods
+
+- *void connectComponents()*: Connects all components into the root of a *javafx.scene.Scene*
+- *void styleComponents()*: Styles all components as required
+- *void setCallbacks()*: Sets callbacks for all interactive components
+
+### *Controllers* package
+
+Only consists of class **SceneManager** used for managing the *javafx* Scenes.
+
+Operations
+
+- *void setStage(final Stage stage)*: Sets the primary stage
+- *void showScene(final Scene scene)*: Replaces the currently active Scene with another one
+- *void showPane(final Class<? extends GamePane> pane)*: Replaces the current  **GamePane** with another
+- *SceneManager getInstance()*: returns the instance of the class
+
+
+## Object diagrams								
+Author(s): `Torstein Thomassen`
+
+This object diagram shows an example of objects with different states in our program.
+All the different attributes and variables has to be fulled in with example data.

```

Comment:
```suggestion
All the different attributes and variables have to be filled in with example data.

```
Made a language suggestion and why is the object diagram not filled in with example data?